### DISCUSSION TITLE
0032 XLS-32d: Request URI Structure

### DISCUSSION THREAD

**Discussion by interc0der** [XRPLF/XRPL-Standards#81] [XLS Proposals]
## The earlier version can be found by looking at the edit history.

**Comment by shortthefomo** [comment]
- Question with regards to how this is exposed. `https://{URL}/{uuid}/{status}` each node now also has to expose https? Are these URI callbacks also to be part of the WSS payloads? How does this fit in with those. Or is that simply and extension that needs to be handled in the py/node libs that are used there.

**Reply by interc0der** [reply]
@lathanbritz 
One of the challenges in standardizing request URI's is transversing messages back to the original application. For example...
Let's say I send a user to your application (threexrp.dev) with a standardized URI string or QR code. After the URI is parsed, validated, and then signed on the ledger through threexrp.dev, ME, as the "issuer" of the URI, might want to know that the URI has been resolved.
In this situation, I would need to set up a webhook or callback so that YOU can let me know when that happens. This could also be implemented through IPFS or pubsub.

It is not the intention of the proposal to implement nodes and web services for every user. Rather, this would be an opt-in feature, and implementation would be the responsibility of the issuer. 

FYI: I have just got around to writing some implementation for this standard. See here -> https://github.com/standardconnect/xls-32d
Let me know what you think =P

**Comment by mvadari** [comment]
@interc0der should this spec be moved into PR and kept in the repo or should it be closed? Note that this does not mean that the spec has been accepted/implemented/finalized.

I will move it to PR by copy-pasting the existing draft if there is no response in the next week.
